### PR TITLE
(PUP-7321) Update REST API docs for Puppet 5

### PR DESCRIPTION
**Pull request by MosesMendoza**
Builds on PR #5990:

> Add application/json as a supported format.
>
> Use text/pson and text/plain consistently.
>
> YAML is no longer accepted.

Includes that commit plus additional changes in response to feedback in that PR.

### CODE REVIEW

**Comment by MosesMendoza** [comment]
Supersedes #5990

File: api/docs/http_api_index.md [hunk 1]
```diff
@@ -170,6 +178,7 @@ Serialization Formats
 Puppet sends messages using several different serialization formats. Not all
 REST services support all of the formats.
 
+* [JSON](https://tools.ietf.org/html/rfc7159)
 * [PSON](./pson.md)
-* [YAML](http://www.yaml.org/spec/1.2/spec.html)
 
+`YAML` was supported in earlier versions of puppet, but is no longer for security reasons.
```
Comment (jtappa) [inline api/docs/http_api_index.md]: capitalize Puppet

File: api/docs/http_catalog.md [hunk 2]
```diff
@@ -34,10 +34,9 @@ The examples below use the POST method.
 Four parameters should be provided to the POST or GET:
 
 - `environment`: the environment name.
-- `facts_format`: must be `pson`.
-- `facts`: serialized pson of the facts hash. One odd note: due to a long-ago misunderstanding in the code, this is
-doubly-escaped (it should just be singly-escaped). To keep backward compatibility, the extraneous
-escaping is still used/supported.
+- `facts_format`: must be `application/json` or `pson`.
+- `facts`: serialized json or pson of the facts hash. Since facts can contain `&`, which
```
Comment (jtappa) [inline api/docs/http_catalog.md]: uppercase JSON and PSON

File: api/docs/http_report.md [hunk 3]
```diff
@@ -21,30 +21,30 @@ PUT
...
-The content of a report is typically generated by the Puppet Runtime and consists of a pson-serialization of `Puppet::Transaction::Report` object which in turn contains a structure of objects with of the following runtime types:
+The content of a report is typically generated by the Puppet Runtime and consists of a json-serialization of `Puppet::Transaction::Report` object which in turn contains a structure of objects with of the following runtime types:
 
 * `Puppet::Util::Log`
 * `Puppet::Util::Metric`
 * `Puppet::Resource::Status`
 * `Puppet::Transaction::Event`
 
-This pson-serialization is compliant with the endpoint's report jsonschema.
+This json-serialization is compliant with the endpoint's report json schema.
```
Comment (jtappa) [inline api/docs/http_report.md]: change to:
`This JSON serialization is compliant with the endpoint's report JSON schema.`

**Comment by MosesMendoza** [comment]
Thanks @jtappa - updated

File: api/docs/http_report.md [hunk 4]
```diff
@@ -29,14 +29,14 @@ None
...
-The content of a report is typically generated by the Puppet Runtime and consists of a pson-serialization of `Puppet::Transaction::Report` object which in turn contains a structure of objects with of the following runtime types:
+The content of a report is typically generated by the Puppet Runtime and consists of a json-serialization of `Puppet::Transaction::Report` object which in turn contains a structure of objects with of the following runtime types:
 
 * `Puppet::Util::Log`
 * `Puppet::Util::Metric`
 * `Puppet::Resource::Status`
 * `Puppet::Transaction::Event`
 
-This pson-serialization is compliant with the endpoint's report jsonschema.
+This json serialization is compliant with the endpoint's report JSON schema.
```
Comment (jtappa) [inline api/docs/http_report.md]: JSON should be uppercase here too please :)

Comment (MosesMendoza) [inline api/docs/http_report.md]: Thanks, updated

**Comment by jtappa** [comment]
👍

**Comment by joshcooper** [comment]
I'll retarget to https://github.com/puppetlabs/puppet/tree/5.0.0-release and merge

**Pull request merged**